sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -343,29 +340,33 @@ private void closeSession() {
         }
 
         @Override
-        public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws Exception {
-            LOGGER.error("Uncaught exception: ", e.getCause());
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+            LOGGER.error("Uncaught exception: ", cause);
         }
 
         @Override
-        public void channelDisconnected(ChannelHandlerContext ctx, ChannelStateEvent e) throws Exception {
+        public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
             LOGGER.trace("channelDisconnected");
             closeSession();
-            super.channelDisconnected(ctx, e);
+            super.channelUnregistered(ctx);
         }
     }
 
-    private void handleStartupBody(ChannelBuffer buffer, Channel channel) {
+    private void handleStartupBody(ByteBuf buffer, Channel channel) {
         Properties properties = readStartupMessage(buffer);
         authenticate(channel, properties);
     }
 
     private void authenticate(Channel channel, Properties properties) {
         String userName = properties.getProperty("user");
-        InetAddress address = CrateNettyHttpServerTransport.getRemoteAddress(channel);
```

Comment:
No I removed it because I did that in a separate commit where the CrateNettyHttpServerTransport was still based on netty3. I'll add a fixup to change it